### PR TITLE
[RFC]: guide: Add a user guide

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add a high level 'read this first' user guide that can try to cover
and/or link out to 'all things kata'.

In my head this is called the 'BigBookOfKata' :-)

Fixes: #311

Signed-off-by: Graham Whaley <graham.whaley@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
/cc @marcov @jodh-intel @kata-containers/architecture-committee 
I've had this idea in my head for a good few weeks now. I drafted out a rough TOC to get an idea what the doc might container. So, RFC - and things to discuss:

1) Is this a sane idea?
2) Shall we do this in a PR'd markdown, or in a wiki
3) if a wiki, would we leave it there, or migrate off to a PR'd markdown later
4) this is pretty rough right now - so I'm not looking for review comments - but a higher level holistic input on if this is something we want and need
5) but, feel free to inject more topics and items for the TOC (and I'll update this RFC).
6) the order of the TOC no doubt will change.

Fundamentally, I found I was struggling to find some documentation when I really needed it, and have a feeling we have gaps and/or things that may only live in gists or slide decks or even just in heads... I'd really like to get everything linked off one place, and easier to find...

**Comment by jodh-intel** [comment]
Hi @grahamwhaley - Nice!

Yes, I think this is a good idea, as long as we are obsessively vigilant to ensure this doc:

- Doesn't duplicate the docs we already have.
- Has as many links as possible to the detailed docs we already have.

Although I still think there is scope for some [wiki-based material from the community](https://github.com/kata-containers/community/wiki), I think this would be better as an in-tree doc as you've created in this PR. The reason being that keeping it here would allow us to fork the doc repo (#246), which I still think we should do. Such forks will allow us to point at the docs *that were used for a particular release* rather than having to handle every possible scenario in our docs (in other words, the docs can be kept as clean and simple as possible).

File: User-Guide.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+common Linux distributions, using the common Linux packaging tools.
+
+
+Details on installation can be found in [documentation repository](https://github.com/kata-containers/documentation/tree/master/install).
+
+For the curious, adventurous, developers or those using a distribution not presently
+supported with pre-built pacakges, Kata Containers can be
+[installed from source](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md). If you are on a distribution that is not presently supported, please feel free
+to reach out to the [community](https://github.com/kata-containers/community#community) to
+discuss adding support. Of course, contributions are most welcome.
```
Comment (jodh-intel) [inline User-Guide.md]: Link to https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md?

File: User-Guide.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
+# Introduction
```
Comment (jodh-intel) [inline User-Guide.md]: A TOC would be bodacious dude! :guitar: :smile:

**Comment by jodh-intel** [comment]
... we could of course do a bit of collaborative doc hacking by hosting this draft in the wiki for a while though... ?

File: User-Guide.md (hunk 2 above)
Comment (grahamwhaley) [reply User-Guide.md]: :-) how predictable you are ;-) yeah, once we have settled on the order of the items then I'll toc it up before first non-RFC PR :-)

Comment (jodh-intel) [reply User-Guide.md]: Yeah, this is "chicken and egg" stuff, but even now it's difficult to see the structure without a TOC. Sigh... would be sooo easy with rst... ;)

**Comment by raravena80** [comment]
@grahamwhaley any updates on pushing this through?

**Comment by grahamwhaley** [comment]
The wiki page version (https://github.com/kata-containers/documentation/wiki/UserGuide) has been filling out, but we've all been a bit busy this month to do any more updates....
I think we need to get some more body and sections of the wiki filled out before we promote it to a markdown file in the repo. So, some progress, but not as much as I'd like and not to the point where we can 'rebase' and merge I think.

**Comment by jodh-intel** [comment]
How's the BBOK coming along @grahamwhaley?

**Comment by raravena80** [comment]
@grahamwhaley any updates on this? (ping)

**Comment by grahamwhaley** [comment]
Hi Rico.
 There have been trickles of input on the relevant wiki page, but there are some glaring big gaps still in there I saw earlier this week.
I can't promise to find a good chunk of time to try and fill some out, or hassle other folks to. I suspect the only way this will get more complete is for a few folks to get together and have a focused effort.

In the meantime, I think we need to leave this here both as a base reference, and to remind/nag folks that the wiki page is sat there asking to be updated :-)

**Comment by raravena80** [comment]
@grahamwhaley monthly ping. 😄

**Comment by raravena80** [comment]
@grahamwhaley updates 😄 🎈 ?

**Comment by raravena80** [comment]
Any updates on this @grahamwhaley ?

**Comment by grahamwhaley** [comment]
Ah, 'the nudge' ;-)
tbh, I'm not getting any time to update the guide, and afaict nobody else updates it either. I'm prone to say we should just shut it down, but make sure any/all info in it is already included in our docs elsewhere..... wdyt @egernst @jodh-intel

**Comment by jodh-intel** [comment]
It might be worth bringing this up at the AC meeting; I think there is a lot of value in such a guide. Writing good technical docs is very time consuming and very hard to get right, but we need this.

Random thought: we could have a "doc sprint" where the entire community hacks on the wiki doc I wonder?

**Comment by ganeshmaharaj** [comment]
I second @jodh-intel 's comment. Do think having one would help, but we should avoid duplicates if that is do-able. (too many docs will kill the dev). 😛

**Comment by grahamwhaley** [comment]
This is going to be very out of date now, and nobody (including me) found time to pull it or push it further... I'll close for now... there is still the copy in the wiki.

**Pull request closed without merging**